### DISCUSSION TITLE
Redirecting output of a command to a file whose name is contained in a variable

### DISCUSSION THREAD

**Discussion by EtaoinWu** [xonsh/xonsh#5463] [Q&A] [answered]
Hi community, I want to do something like this:

```xonsh
x = '1.txt'
ls > @(x)
```

However, I got the following output:

```
xonsh: ['1.txt']: unable to open file
```

So my question is: how can I redirect a command's stdout to a variable-named file?

**Comment by anki-code** [comment, marked as answer]
Hey! This fixed in #5423 please use current version from main branch i.e.
```xsh
xpip install --force-reinstall 'git+https://github.com/xonsh/xonsh'
# restart xonsh
```
